### PR TITLE
Update the design of the command center

### PR DESCRIPTION
**Pull request by youknowriad**
Related to https://github.com/WordPress/gutenberg/issues/48457
Alternative to #49658 

## What?

This PR updates the design of the command center per @jasmussen's mockups:

 - Tweaks the radius, borders...
 - Removes "groups" between the different sections (posts, pages, templates...)
 - Add icons to each command
 - Results only available when the input is not empty

**Note** I'm seeing a weird design glitch on Safari when I search quickly. Not sure exactly where it comes from.

## Testing Instructions

1- Open the experiments page
2- Enable the command center
3- Navigate to the site editor
4- Hit cmd+k and try using the command center.

### CODE REVIEW

**Comment by jasmussen** [comment]
Thanks for taking a stab!

I realize this keeps coming up for me, but I must be doing something wrong with my environment, I'm not seeing any difference in this branch:

Screenshot 2023-04-10 at 12 29 39

> Tweaks the radius

A note that this may be best done separately, as that radius could/should be the new default radius for modals, so not just unique to this one: 

Screenshot 2023-04-10 at 12 31 16

**Comment by youknowriad** [comment]
@jasmussen yeah that's not correct, would you mind trying to restart the build?

**Comment by jasmussen** [comment]
For some reason, that worked, though I'm sure I tried this before. GIF showing the new:

testing

A few small notes. The gray highlight I in the mockup, I have from the URL dialog:

Screenshot 2023-04-10 at 12 49 53

No strong opinion on whether that's gray or blue, but would prefer we unify across those.

A small detail, I search for "Con" and it finds first "Comments". It's fine that it finds that through the fuzzy stuff, but as you can see in the GIF, it keeps highlighting that. I feel like it should always highlight the _top_ item, at least until you move the arrow keys. What do you think?

The bold highlight looks great, nice.

The icon is 16x16, it should be 24x24:

Screenshot 2023-04-10 at 12 52 31

The search field font is 16px in this branch, I bumped it all the way to 20px in my mockup. This is not a strong opinion, but it makes it feel more substantial to me. I'd love a broader design opinion on this.

Finally, there's a question of what the placeholder text should be. In my mockup I have some fairly boilerplate text:

> Type a command or search

The good thing about that is that it's _short!_ That allows for a less wide modal, makes it feel more manageable and friendly in a way. It's 400px wide in the mockups, we can go wider than that, but it feels a bit too wide in this branch as is. Speaking of, this branch says:

> Search for content and templates, or try commands like "Add…"

In principle, the tip is useful, but I wonder:

* Can we compress that sentence?
* Or can we rotate an array of tips? I.e. it might say "Search for content and templates" one time, and "Try commands like Add" in another? I'm not sure it's the best idea especially here at the start, but it might be a good way to surface the diversity of actions you can take, without too long of a sentence.

A single unified alternative to start with might be something like:

> Search or try commands like "Add"

Mainly this is driven by the instinct: the less text, the more likely it is to be read. Similar to our:

> Type / to choose a block

What do you think?

**Comment by youknowriad** [comment]
I also like the short placeholder personally. and I'm going to update here with the design feedback only. The thing about searches... is still unclear and I'd prefer to address separately.

**Comment by youknowriad** [comment]
I think I've addressed all the design feedback.

**Comment by jasmussen** [comment]
Looks great to me:

current status

Upon second thought, perhaps the search field could be slightly wider, perhaps a nice round 480px max-width:

Screenshot 2023-04-10 at 13 42 48

I also appreciate that the top edge of the modal stays fixed even as it grows vertically. 

I think there are lots of little things we can improve here, and I'd love feedback from @richtabor specifically as he has separte mockups for this. I imagine many are back to review this tomorrow. What do you think?

**Comment by youknowriad** [comment]
400px felt too small indeed. I've update the width and I'm happy to wait for more feedback.

**Comment by alexstine** [comment]
I do not recall this modal having a close button. Can you work it into the design? It does need one. I will be able to do a more solid review of this once the related shortcut PR lands and makes this easier to access on Windows.

**Comment by jameskoster** [comment]
> No strong opinion on whether that's gray or blue, but would prefer we unify across those.

So far as I've seen the trend is towards the blue. E.g. Inserter items, tertiary buttons, list view. The gray feels 'old' to me.

I'm not a fan of using icons as the _only_ differentiator. How will we communicate what is a post category vs a product category if they appear together in search results? We don't necessarily need to solve this here, but it's going to come up as soon as we add support for taxonomies.

**Comment by Mamaduka** [comment]
> Note I'm seeing a weird design glitch on Safari when I search quickly. Not sure exactly where it comes from.

@youknowriad, is the input value debounced? In that case, the issue might have been resolved by #49234.

We still need to override the core polyfill and use one generated by the plugin.

**Comment by youknowriad** [comment]
@Mamaduka no, I don't think so, also your link is wrong :P

**Comment by Mamaduka** [comment]
@youknowriad, I'm out of the practice copy-pasting issues 😄 I fixed the link.

**Comment by youknowriad** [comment]
@Mamaduka so you're saying that once we "override" core polyfills by the Gutenberg plugin, the issue will fix itself?

**Comment by Mamaduka** [comment]
I can't say for sure, but we still need to override the core polyfill so that fix proposed by @sgomes is adequately tested.

P.S. We have a similar report for the block inserter search and Safari - #49208.

**Comment by sgomes** [comment]
> In that case, the issue might have been resolved by #49234.

I should point out that while it's plausible that #49234 would help with performance, I wasn't able to measure any impact while working on that change, and ended up submitting it more out of cleanliness and removing unnecessary code than anything else.

If you are able to measure a performance improvement, though, please do let me know, as I'd love to know more!

**Comment by richtabor** [comment]
> I think there are lots of little things we can improve here, and I'd love feedback from @richtabor specifically as he has separte mockups for this. 

It input font size feels quite bit, especially in relation to the menu items. It's one of the largest font sizes within the editor. Perhaps dropping it down to `18px` tones it a bit? 

CleanShot 2023-04-14 at 17 06 17
CleanShot 2023-04-14 at 17 07 08

And units around the input feel off when there are results (too much space above): 
CleanShot 2023-04-14 at 17 09 09

We could probably reduce the padding on the items, which would better align with the input text as well: 

CleanShot 2023-04-14 at 17 10 29

**Comment by jameskoster** [comment]
I pushed some updates based on the [Figma](https://www.figma.com/file/8SnuEfNOES3X07TAGyukd2/Command-Center?node-id=0%3A1&t=RqcMOs7ROBvN2cNQ-1) we've been working in:

Screenshot 2023-04-17 at 14 14 16

It would be nice to include a 'Type' chip in each item so that you can easily tell what's what – the icon isn't always obvious. 

Screenshot 2023-04-17 at 14 18 19

Probably fine to do that in a follow-up though :)

**Comment by jasmussen** [comment]
I was thinking the radius change should happen separately, it likely needs a README update. But I don't mind bundling it in here if you all are okay with it.

Maybe time to switch to the modern color scheme? :D

**Comment by jameskoster** [comment]
> Maybe time to switch to the modern color scheme? :D

Maybe 🤣

The radius is only applied to the command center. Agree it would be nice to update all modals though, and that should happen separately. I guess I'll revert that here.

**Comment by youknowriad** [comment]
Are we ready to ship this for now?

**Review by jasmussen — APPROVED** [review]
This feels like a solid step in the right direction. Let's land this as it's a strong improvement over what's shipping. Here's a GIF showing the commandbar, searching for "con" and finding the "Contacto" page, searching for "add" and finding "add new post", pressing Enter to navigate there, and it working as intended:

commandbar

The visuals will look stronger still when #49870 lands.

One gotcha, which is more of a question. If I search for `add ` (including the space) I would expect to see results for both "add new post" and "add new page", but I get that only for `add`. This isn't critical to get right, but it would be a nice optimization if we could, because it would avoid a weird "no results" flicker if I'm typing out "add new post" as my command.

**Comment by youknowriad** [comment]
Yeah, search is a bit weird (it's internal to the library we use) but I'm going to see how we can improve that.

**Pull request merged**
